__tests__/components/editor/ImportResourceTemplate.test.js

```diff
@@ -124,44 +124,125 @@ describe('<ImportResourceTemplate />', () => {
 
       expect(wrapper.state().modalShow).toBe(false)
 
       wrapper.instance().updateStateFromServerResponse({ status: 409 , headers: {} }, 0)
+
       expect(wrapper.state().modalShow).toBe(true)
     })
+
     it('sets message in state with any create operation not resulting in HTTP 409', () => {
       // Set new wrapper in each of these tests because we are changing state
       wrapper = shallow(<ImportResourceTemplate.WrappedComponent authenticationState={authenticationState}/>)
 
       expect(wrapper.state().message).toEqual([])
 
-      wrapper.instance().updateStateFromServerResponse({ status: 201 , headers: { location: 'http://foo.bar' } })
+      wrapper.instance().updateStateFromServerResponse({ status: 201 , headers: { location: 'http://foo.bar' } }, 0)
 
       expect(wrapper.state().message).toEqual(['Created http://foo.bar'])
     })
+
+    it('resets createResourceError and message in component state when response is OK', () => {
+      // Set new wrapper in each of these tests because we are changing state
+      wrapper = shallow(<ImportResourceTemplate.WrappedComponent authenticationState={authenticationState}/>)
+
+      wrapper.setState({ createResourceError: ['foo'], message: ['bar'] })
+      expect(wrapper.state().createResourceError).toEqual(['foo'])
+      expect(wrapper.state().message).toEqual(['bar'])
+
+      wrapper.instance().updateStateFromServerResponse({ ok: true, status: 204 }, 0)
+
+      expect(wrapper.state().createResourceError).toEqual([])
+      expect(wrapper.state().message).toEqual(['Updated '])
+    })
+
+    it('leaves createResourceError alone and appends message in component state when response is not OK', () => {
+      // Set new wrapper in each of these tests because we are changing state
+      wrapper = shallow(<ImportResourceTemplate.WrappedComponent authenticationState={authenticationState}/>)
+
+      wrapper.setState({ createResourceError: ['foo'], message: ['bar'] })
+      expect(wrapper.state().createResourceError).toEqual(['foo'])
+      expect(wrapper.state().message).toEqual(['bar'])
+
+      wrapper.instance().updateStateFromServerResponse({ ok: false, status: 400 }, 0)
+
+      expect(wrapper.state().createResourceError).toEqual(['foo'])
+      expect(wrapper.state().message).toEqual(['bar', 'Unexpected response (400)! '])
+    })
+  })
+
+  describe('humanReadableLocation()', () => {
+    it('returns human-readable label when Location header exists', () => {
+      const response = {
+        headers: {
+          location: 'http://foo.bar/baz'
+        },
+        req: {
+          url: 'http://foo.bar',
+          _data: {
+            id: 'baz'
+          }
+        }
+      }
+      expect(wrapper.instance().humanReadableLocation(response)).toEqual('http://foo.bar/baz')
+    })
+
+    it('returns human-readable label when data ID exists', () => {
+      const response = {
+        req: {
+          url: 'http://foo.bar',
+          _data: {
+            id: 'baz'
```

Comment:
maybe "baz" could be 'data_id' ?  and url could be 'http://example.org'?  I'm getting confused as to what "label" is and what is constituted into it.  (applies to other tests using foo and bar)

Reply:
I'll rework this.